BookingsApi/BookingsApi/Controllers/HearingsController.cs

```diff
@@ -795,11 +788,37 @@ public async Task<IActionResult> GetUnallocatedHearings()
 
             if (results.Count <= 0)
                 _logger.TrackEvent("[GetUnallocatedHearings] Could not find any unallocated hearings");
-            var hearingMapper = new HearingToDetailsResponseMapper();
-            var response = results.Select(hearingMapper.MapHearingToDetailedResponse).ToList();
+            var response = results.Select(HearingToDetailsResponseMapper.Map).ToList();
             return Ok(response);
         }
+ 
+        /// <summary>
+        /// Search for hearings to be allocate via search parameters
+        /// </summary>
+        /// <param name="searchRequest">Search criteria</param>
+        /// <returns>list of hearings matching search criteria</returns>
+        [HttpGet("allocation/search", Name = "SearchForAllocationHearings")]
+        [OpenApiOperation("SearchForAllocationHearings")]
+        [ProducesResponseType(typeof(List<HearingDetailsResponse>), (int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        public async Task<IActionResult> SearchForAllocationHearings([FromQuery] SearchForAllocationHearingsRequest searchRequest)
+        {
+
+            var query = new GetAllocationHearingsBySearchQuery(
+                searchRequest.CaseNumber, 
+                searchRequest.CaseType, 
+                searchRequest.FromDate, 
+                searchRequest.ToDate, 
+                searchRequest.CsoUserName);
+            
+            var hearings = await _queryHandler.Handle<GetAllocationHearingsBySearchQuery, List<VideoHearing>>(query);
 
+            if (hearings == null || !hearings.Any())
+                return NotFound();
```

Comment:
could we return an empty list rather than a not found error? Not found errors tend to be for bad urls, normally implying the user has requested an incorrect route.

the other "list" endpoints in this controller seem to follow a similar pattern.

Reply:
Yeah it's something I've not quite made mind up on - as to what the right way is. I see it debated alot online aswell. With some people suggesting 204 for no data found https://softwareengineering.stackexchange.com/questions/203492/when-to-use-http-status-code-404-in-an-api 

In other parts of the code this is how it's handled with a 404. But I think you're right and an empty 200 is the way to go to be consistent with the rest of the hearings controller.

Reply:
sounds good :)

yeah there's no right or wrong answer. I tend to return 200s for lists, 404s if the actual resource specified as part of the url does not exist, eg GET users/123

https://stackoverflow.com/questions/13366730/proper-rest-response-for-empty-table#:~:text=The%20404%20status%20code%20should,exists%20but%20it's%20currently%20empty.